maven/main/java/bsu/fpmi/kwitter/TweetControlServlet.java

```diff
@@ -0,0 +1,149 @@
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        String id = req.getParameter("id");
+        PrintWriter writer = resp.getWriter();
+        resp.setContentType("application/json");
+
+        if (id == null) {
+            id = NONEXISTENT_POST_ID;
+        }
+
+        Gson gson = new Gson();
+        writer.write(gson.toJson(posts.getById(Integer.parseInt(id))));
+        writer.close();
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        Gson gson = new Gson();
+        StringBuilder stringBuilder = new StringBuilder();
+        BufferedReader bufferedReader = req.getReader();
+        String line;
+        while ((line = bufferedReader.readLine()) != null) {
+            stringBuilder.append(line);
+        }
+        System.out.println(stringBuilder.toString());
+        Post post;
+        try {
+           post = gson.fromJson(stringBuilder.toString(), new TypeToken<Post>(){}.getType());
+        } catch (com.google.gson.JsonSyntaxException e) {
+            e.printStackTrace();
+            return;
+        }
+
+
+        PrintWriter writer = resp.getWriter();
+
+        HttpServletRequest httpRequest = (HttpServletRequest)req;
+
+        String id = req.getParameter("id");
+        if (id == null) {
+            id = NONEXISTENT_POST_ID;
+        }
+
+        String userName = req.getParameter("userName");
+
+        switch (httpRequest.getServletPath()) {
+            case "/tweets/search": {
+                PostCollection filterResult = posts.getPosts(post);
+                for (int i = 0; i < filterResult.size(); ++i) {
+                    writer.write(gson.toJson(filterResult.get(i)));
+                }
+                break;
+            }
+
+            case "/tweets/edit": {
+                writer.write(gson.toJson(posts.editById(Integer.parseInt(id), post)));
+            }
+
+            case "/tweets/like": {
+                boolean result;
+                if (userName == null) {
+                    result = false;
+                } else {
+                    result = posts.likePost(Integer.parseInt(id), userName);
+                }
+                writer.write(gson.toJson(result));
+                break;
+            }
+
+            case "/tweets/unlike": {
+                boolean result;
+                if (userName == null) {
+                    result = false;
+                } else {
+                    result = posts.unlikePost(Integer.parseInt(id), userName);
+                }
+                writer.write(gson.toJson(result));
+                break;
+            }
+
+            case "/tweets/getAll": {
+                for (int i = 0; i < posts.size(); ++i) {
+                    writer.write(gson.toJson(posts.get(i)));
+                }
+                break;
+            }
+
+            default: {
```

Comment:
Such default is useless. But then, to be consistent, it would make sense to set , for example, status 404 for a default, of print out some message. Because here it means "this address cannot be processed", doesn't it?